src/local-history/local-history-manager.ts

```diff
@@ -139,38 +143,40 @@ export class LocalHistoryManager {
-        if (!fs.existsSync(hashedFolderPath)) {
-            fs.mkdirSync(hashedFolderPath, { recursive: true });
-        }
+        if (this.fileSizeLimit(document)) {
+            const timestamp = this.getCurrentTime();
+            const timestampForFileName = timestamp.replace(/[-:. ]/g, '');
+            const fileFullPath = path.parse(document.fileName);
+            const historyFileName = `${fileFullPath.name}_${timestampForFileName.substring(0, 14)}_${timestampForFileName.substring(14, 17)}${fileFullPath.ext}`;
+            const hashedFolderPath = this.getHashedFolderPath(document.fileName);
+
+            // Create a folder (and all the parent folders) for storing all the local history file for the active editor.
+            if (!fs.existsSync(hashedFolderPath)) {
+                fs.mkdirSync(hashedFolderPath, { recursive: true });
+            }
 
-        try {
-            const historyFilePath = path.join(hashedFolderPath, historyFileName);
-            // Copy the content of the current active editor.
-            const activeDocumentContent: string = await this.readFile(document.fileName);
-            const recentUpdatedFile = this.getMostRecentRevision(hashedFolderPath);
-            if (recentUpdatedFile) {
-                const latestEditorHistoryContent: string | undefined = recentUpdatedFile && await this.readFile(recentUpdatedFile);
-                if (this.historyFileTimeDifference(document) && latestEditorHistoryContent && !(activeDocumentContent === latestEditorHistoryContent)) {
-                    await this.writeFile(historyFilePath, activeDocumentContent, true);
-                    return;
-                }
-                else {
-                    fs.renameSync(recentUpdatedFile, historyFilePath);
-                    await this.writeFile(historyFilePath, activeDocumentContent, true);
-                    return;
+            try {
+                const historyFilePath = path.join(hashedFolderPath, historyFileName);
+                // Copy the content of the current active editor.
+                const activeDocumentContent: string = await this.readFile(document.fileName);
+                const recentUpdatedFile = this.getMostRecentRevision(hashedFolderPath);
+                if (recentUpdatedFile) {
+                    const latestEditorHistoryContent: string | undefined = recentUpdatedFile && await this.readFile(recentUpdatedFile);
+                    if (this.historyFileTimeDifference(document) && latestEditorHistoryContent && !(activeDocumentContent === latestEditorHistoryContent)) {
+                        await this.writeFile(historyFilePath, activeDocumentContent, true);
+                        return;
+                    }
+                    else {
+                        fs.renameSync(recentUpdatedFile, historyFilePath);
+                        await this.writeFile(historyFilePath, activeDocumentContent, true);
+                        return;
+                    }
```

Comment:
I don't quite understand the need to re-format all this code.
The check for the size can be quick:

```ts
if (!this.filesizeLimit(document) {
   return;
}
```